public/app/core/components/AppChrome/ReturnToPrevious/ReturnToPrevious.tsx

```diff
@@ -0,0 +1,62 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2, UrlQueryValue } from '@grafana/data';
+import { locationService } from '@grafana/runtime';
+import { Button, ButtonGroup, useStyles2 } from '@grafana/ui';
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
+
+export interface ReturnToPreviousProps {
+  href: UrlQueryValue;
+  title: UrlQueryValue;
+  children: UrlQueryValue;
+}
+
+export const ReturnToPrevious = ({ href, title, children }: ReturnToPreviousProps) => {
+  const styles = useStyles2(getStyles);
+  const [, setParams] = useQueryParams();
+
+  const handleOnClick = () => {
+    href && locationService.push(href.toString());
+  };
+  const closeButton = () => {
+    setParams({ __returnToTitle: null, __returnToUrl: null });
+    const currentLocation = locationService.getLocation();
+    locationService.push(currentLocation);
```

Comment:
This creates a strange pattern to me, when I press the 'x' and then press the browser back button it keeps the same URL, one more back and it readds the params to the URL. I think we need to investigate this and maybe use `locationService.replace`, but I would double check what is the intended UX here.